blockchain.go

```diff
@@ -1112,11 +1119,26 @@ func convertToSealedResults(
 	return output, nil
 }
 
-// debugErrorHashingAlgo tries to unwrap error to the root and test for invalid hashing algorithms
-func (b *Blockchain) debugErrorHashingAlgo(err error, tx *flowgo.TransactionBody) error {
+// testGeneralSignatureError checks if there was signature error and add context for debugging
+func (b *Blockchain) testGeneralSignatureError(err error, tx *flowgo.TransactionBody) error {
+	flowErr, ok := err.(*types.FlowError)
+	if !ok {
+		return nil
+	}
+
+	_, ok = flowErr.Unwrap().(*fvmerrors.InvalidProposalSignatureError)
```

Comment:
Could `errors.Is` or `errors.As` be used here? https://pkg.go.dev/errors

Reply:
good idea, thanks